diff --git a/src/synfigapp/blineconvert.cpp b/src/synfigapp/blineconvert.cpp
index c954d8d3f..b3b3465d1 100644
--- a/src/synfigapp/blineconvert.cpp
+++ b/src/synfigapp/blineconvert.cpp
@@ -560,7 +560,7 @@ synfigapp::BLineConverter::operator()(std::list<synfig::BLinePoint>  &blinepoint
 
 		cum_dist.resize(point_cache.size()); this_dist.resize(point_cache.size());
 		Real d = 0;
-		for(unsigned int i = 0; i < point_cache.size();)
+		for(unsigned int i = 0; i < point_cache.size() - 1;)
 		{
 			d += (this_dist[i] = (p2-p1).mag());
 			cum_dist[i] = d;
